### DISCUSSION TITLE
Show name of a related entity in disabled text input

### DISCUSSION THREAD

**Discussion by kolaente** [filamentphp/filament#12906] [Help] [answered]
### Package

Panel builder

### Package Version

3.2.66

### How can we help you?

In my form (using the panel builder), I have the following field:

```php
Forms\Components\TextInput::make('orderedBy.name')
    ->disabled()
    ->label(__('Ordered by'))
    ->hintAction(fn (Order $order) => $order->orderedBy ? Forms\Components\Actions\Action::make(__('Send mail'))
        ->icon('heroicon-s-at-symbol')
        ->url(
            "mailto:{$order->orderedBy->email}",
        ) : null,
    ),
```

where `orderedBy` is a relation defined on the model like this:

```php
public function orderedBy(): BelongsTo
{
    return $this->belongsTo(User::class);
}
```

However, the input field is always empty.

I've also tried to create a custom attribute like this:

```php
public function orderedByNameField(): Attribute
{
    return Attribute::make(
        get: fn() => $this->orderedBy->name,
    );
}
```

then referenced it as `ordered_by_name_field` in the text input, but without any success.

How can I make this work?

**Comment by leandrocfe** [comment, marked as answer]
Shoudn't it be a Select?
https://filamentphp.com/docs/3.x/forms/fields/select/#integrating-with-an-eloquent-relationship

**Reply by kolaente** [reply]
That works. Thanks!
